test/conversions.jl

```diff
@@ -45,3 +45,37 @@ end
         @test float32Val == Float32sr(float16Val)
     end
 end
+
+
+
+for SR in [Float16sr, BFloat16sr, Float32sr]
+
+    # corresponding FP type
+    FP = float(SR)
+
+    @eval begin
+        @testset "Stochastic conversion of Float64 to $(nameof($SR))" begin
+            for k in 1:100
+                trueVal = randn(Float64)
+
+                # make sure that this is not representable by FP
+                if trueVal == $FP(trueVal)
+                    trueVal = nextfloat(trueVal)
+                end
+
+                roundedVal = $SR(trueVal)
+
+                # check that rounding is not deterministic
+                is_stochastic = false
+                for l in 1:10000
+                    if roundedVal != $SR(trueVal)
+                        is_stochastic = true
+                        break
+                    end
+                end
+
+                @test is_stochastic == true
```

Comment:
No need to check that a boolean is equal to another boolean, you can just test its boolean value 🙂 
```suggestion
                @test is_stochastic
```

Reply:
Yes, I know. I found it more readable as a user in this particular scenario.
A failing test is more semantic since we get the message

```
  Expression: is_stochastic == true
   Evaluated: false == true
```

instead of 
```
  Expression: is_stochastic
```

I think it is better in the first variant.

But I can change it to the second one if you want.

Reply:
no need, that's fine by me and I agree with your point